packages/web-client/tests/integration/components/card-pay/create-merchant-workflow/prepaid-card-choice-test.ts

```diff
@@ -84,26 +103,115 @@ module(
+      await click(`[data-test-card-picker-dropdown-option="${cardAddress}"]`);
+    }
+
+    test('it shows the correct data in default state', async function (assert) {
+      assert
+        .dom(`[data-test-boxel-card-container]`)
+        .containsText('Choose a prepaid card to pay the merchant creation fee');
       assert
         .dom('[data-test-prepaid-card-choice-merchant-fee]')
         .containsText('100 SPEND');
       assert
         .dom('[data-test-merchant]')
-        .hasAttribute('data-test-merchant', "It's a new merchant")
-        .hasAttribute('data-test-merchant-logo-background', '#ffffff')
-        .hasAttribute('data-test-merchant-logo-text-color', '#000000');
+        .hasAttribute('data-test-merchant', 'Mandello')
+        .hasAttribute('data-test-merchant-logo-background', '#ff5050')
+        .hasAttribute('data-test-merchant-logo-text-color', '#fff');
+      assert
+        .dom('[data-test-prepaid-card-choice-merchant-id]')
+        .containsText('mandello1');
+      assert
+        .dom(`[data-test-boxel-action-chin] [data-test-boxel-button]`)
+        .isDisabled();
+      assert
+        .dom(`[data-test-card-picker-dropdown]`)
+        .containsText('Select card');
+
+      await click(`[data-test-card-picker-dropdown] > [role="button"]`);
+      await waitFor(
+        `[data-test-card-picker-dropdown-option="${prepaidCardAddress}"]`
+      );
+
+      assert
+        .dom(`[data-test-card-picker-dropdown-option]`)
+        .exists({ count: 2 });
+
+      await click(
+        `[data-test-card-picker-dropdown-option="${prepaidCardAddress}"]`
+      );
+
+      assert
+        .dom(
+          `[data-test-prepaid-card-choice-selected-card] [data-test-prepaid-card]`
+        )
+        .exists();
+      assert
+        .dom(`[data-test-card-picker-dropdown]`)
+        .containsText('Change card');
+      assert
+        .dom(`[data-test-boxel-action-chin] [data-test-boxel-button]`)
+        .isNotDisabled();
+    });
+
+    test('it allows changing selected prepaid card', async function (assert) {
+      await selectPrepaidCard(prepaidCardAddress);
+
+      assert
+        .dom(
+          `[data-test-prepaid-card-choice-selected-card] [data-test-prepaid-card="${prepaidCardAddress}"]`
+        )
+        .exists();
+      assert
+        .dom(`[data-test-prepaid-card-choice-selected-card]`)
+        .containsText(prepaidCardAddress);
+      assert
+        .dom(`[data-test-prepaid-card-choice-selected-card]`)
+        .containsText('2324 SPEND');
+
+      await selectPrepaidCard(prepaidCardAddress2);
+
+      assert
+        .dom(
+          `[data-test-prepaid-card-choice-selected-card] [data-test-prepaid-card="${prepaidCardAddress2}"]`
+        )
+        .exists();
+      assert
+        .dom(`[data-test-prepaid-card-choice-selected-card]`)
+        .containsText(prepaidCardAddress2);
+      assert
+        .dom(`[data-test-prepaid-card-choice-selected-card]`)
+        .containsText('500 SPEND');
+    });
```

Comment:
Nice tests! One more I’d like to see is something that asserts that `registerMerchant` is called with the address of the chosen card; you can see an example of that kind of assertion [here](https://github.com/cardstack/cardstack/blob/4b99e277936acd394149cb9dc52580f6d3dd987a/packages/web-client/tests/integration/components/card-pay/deposit-workflow/transaction-amount-test.ts#L77-L99), elsewhere in this file there are tests that stub the service call.

It’s true that the acceptance test verifies this implicitly but I think it would be nice to see an explicit verification 🤓

Reply:
Added this (starts at line 246 below)

Reply:
excellent, thanks!